content/post/gke-mariadb/index.md

```diff
@@ -0,0 +1,540 @@
+backupsessions.stash.appscode.com         2022-01-26T09:27:23Z
+functions.stash.appscode.com              2022-01-26T09:26:48Z
+recoveries.stash.appscode.com             2022-01-26T09:27:23Z
+repositories.stash.appscode.com           2022-01-26T09:27:23Z
+restics.stash.appscode.com                2022-01-26T09:27:22Z
+restorebatches.stash.appscode.com         2022-01-26T09:27:23Z
+restoresessions.stash.appscode.com        2022-01-26T09:27:23Z
+tasks.stash.appscode.com                  2022-01-26T09:26:48Z
+```
+
+
+### Step 2: Prepare Backend
+
+Stash supports various backends for storing data snapshots. It can be a cloud storage like GCS bucket, AWS S3, Azure Blob Storage etc. or a Kubernetes persistent volume like HostPath, PersistentVolumeClaim, NFS etc.
+
+For this tutorial we are going to use gcs-bucket. You can find other setups [here](https://stash.run/docs/v2021.11.24/guides/latest/backends/overview/).
+
+ ![My Empty GCS bucket](GoogleCloudEmpty.png)
+
+At first we need to create a secret so that we can access the gcs bucket. We can do that by the following code:
+
+```bash
+$ echo -n 'YOURPASSWORD' > RESTIC_PASSWORD
+$ echo -n 'YOURPROJECTNAME' > GOOGLE_PROJECT_ID
+$ cat /PATH/TO/JSONKEY.json > GOOGLE_SERVICE_ACCOUNT_JSON_KEY
+$ kubectl create secret generic -n demo gcs-secret \
+        --from-file=./RESTIC_PASSWORD              \
+        --from-file=./GOOGLE_PROJECT_ID            \
+        --from-file=./GOOGLE_SERVICE_ACCOUNT_JSON_KEY
+ ```
+
+### Step 3: Create Repository
+
+```yaml
+apiVersion: stash.appscode.com/v1alpha1
+kind: Repository
+metadata:
+  name: gcs-repo
+  namespace: demo
+spec:
+  backend:
+    gcs:
+      bucket: stash-backup-dipta
+      prefix: /demo/mariadb/sample-mariadb
+    storageSecretName: gcs-secret
+```
+
+This repository CRD specifies the gcs-secret we created before and stores the name and path to the gcs-bucket. It also specifies the location in the bucket where we want to backup our database.
+> Here, My bucket name is stash-backup-dipta. Don't forget to change `spec.backend.gcs.bucket` to your bucket name.
+
+### Step 4: Create BackupConfiguration
+
+Now we need to create a BackupConfiguration file that specifies what to backup, where to backup and when to backup.
+
+```yaml
+apiVersion: stash.appscode.com/v1beta1
+kind: BackupConfiguration
+metadata:
+  name: sample-mariadb-backup
+  namespace: demo
+spec:
+  schedule: "*/5 * * * *"
+  repository:
+    name: gcs-repo
+  target:
+    ref:
+      apiVersion: appcatalog.appscode.com/v1alpha1
+      kind: AppBinding
+      name: sample-mariadb
+  retentionPolicy:
+    name: keep-last-5
+    keepLast: 5
+    prune: true
+```
+
+* BackupConfiguration creates a cronjob that backs up the specified database (`spec.target`) every 5 minutes.
+* `spec.repository` contains the secret we created before called `gcs-secret`.
+* `spec.target.ref` contains the reference to the appbinding that we want to backup. So, after 5 minutes we can see the following status:
```

Comment:
- To learn more about AppBinding, here [https://kubedb.com/docs/v2021.12.21/guides/mariadb/concepts/appbinding/]
- Add a new line before `So, after 5 minutes we can ...`